test/parallel/test-http2-compat-serverresponse-end.js

```diff
@@ -1,22 +1,28 @@
 // Flags: --expose-http2
 'use strict';
 
+const { strictEqual } = require('assert');
 const { mustCall, mustNotCall } = require('../common');
-const { createServer, connect } = require('http2');
-
-// Http2ServerResponse.end
+const {
+  createServer,
+  connect,
+  constants: {
+    HTTP2_HEADER_STATUS,
+    HTTP_STATUS_OK
+  }
+} = require('http2');
 
 {
-  const server = createServer();
-  server.listen(0, mustCall(() => {
-    const port = server.address().port;
-    server.once('request', mustCall((request, response) => {
-      response.on('finish', mustCall(() => {
-        server.close();
-      }));
-      response.end();
+  // Http2ServerResponse.end callback is called only the first time,
+  // but may be invoked repeatedly without throwing errors.
+  const server = createServer(mustCall((request, response) => {
+    response.end(mustCall(() => {
+      server.close();
     }));
-
+    response.end(mustNotCall());
+  }));
+  server.listen(0, mustCall(() => {
+    const {port} = server.address();
```

Comment:
Minor style nit: `{ port } = ` .... house style is to include spaces around the brackets.